e2e/run_test.go

```diff
@@ -201,3 +201,32 @@ func TestRunWildcardCountGreaterEqualThanWorkerCount(t *testing.T) {
 
 	assertLines(t, result.Stderr(), map[int]compareFunc{})
 }
+
+func TestRunSpecialCharactersInPrefix(t *testing.T) {
+	t.Parallel()
+
+	bucket := s3BucketFromTestName(t)
+
+	s3client, s5cmd, cleanup := setup(t)
+	defer cleanup()
+
+	createBucket(t, s3client, bucket)
+	putFile(t, s3client, bucket, `special-chars_!@#$%^&_()_+{[_%5Cäè| __;'_,_._-中文 =/_!@#$%^&_()_+{[_%5Cäè| __;'_,_._-中文 =image.jpg`, "content")
```

Comment:
You should keep a variable for the file name and use it on `putFile`, `assertLines` and `equals`.

Reply:
Good call - addressed in https://github.com/peak/s5cmd/pull/111/commits/61d38e84ddc0f5ae6743390a1e44af18f9401f15